contracts/Registries/Overrides.sol

```diff
@@ -0,0 +1,118 @@
+    address[] public poolAddresses;
+
+    constructor(
+        address _curveAddressesProviderAddress,
+        address _curveRegistryAddress
+    ) {
+        require(
+            _curveAddressesProviderAddress != address(0),
+            "Missing Curve Addresses Provider address"
+        );
+        require(
+            _curveRegistryAddress != address(0),
+            "Missing Curve Registry address"
+        );
+        curveAddressesProvider = ICurveAddressesProvider(
+            _curveAddressesProviderAddress
+        );
+        curveRegistry = ICurveRegistry(_curveRegistryAddress);
+    }
+
+    /// @notice Updates Curve Registry Address
+    function addCurveRegistry(address _curveRegistryAddress) public onlyOwner {
+        require(
+            _curveRegistryAddress != address(0),
+            "Missing Curve Registry Address"
+        );
+        curveRegistry = ICurveRegistry(_curveRegistryAddress);
+    }
+
+    /// @notice Returns both override and curve registry pools as list
+    function curveRegistriesList() public view returns (address[] memory) {
+        uint256 poolOverrideCount = poolAddresses.length;
+        uint256 poolRegistryCount = curveRegistry.pool_count();
+        uint256 numRegistries = poolOverrideCount + poolRegistryCount;
+        address[] memory _registries = new address[](numRegistries);
+        for (uint256 i; i < poolOverrideCount; i++) {
+            _registries[i] = poolAddresses[i];
+        }
+        for (uint256 i; i < poolRegistryCount; i++) {
+            _registries[i + poolOverrideCount] = curveRegistry.pool_list(i);
+        }
+        return _registries;
+    }
+
+    /// @notice Adds an override pool address for an LP
+    /// @dev Maintains an additional pool address list for indexing
+    function setPoolForLp(address _poolAddress, address _lpAddress)
+        public
+        onlyOwner
+    {
+        poolByLpOverride[_lpAddress] = _poolAddress;
+        poolAddresses.push(_poolAddress);
+    }
+
+    /// @notice Search through pool registry overrides and curve registries for a LP Pool
+    function poolByLp(address _lpAddress) public view returns (address) {
+        address pool = poolByLpOverride[_lpAddress];
+        if (pool != address(0)) {
+            return pool;
+        }
+        pool = executeSelectorOnCurveRegistries(_lpAddress);
+        return pool;
+    }
+
+    /// @notice Cycle through all curve registries and try to staticcall each one using manual selector
+    /// @dev return type is inconsistent across curve registres
+    function executeSelectorOnCurveRegistries(address _lpAddress)
+        public
+        view
+        returns (address)
+    {
+        uint256 numRegistries = curveAddressesProvider.max_id() + 1;
+        address curveRegistryAddress;
+        address pool;
+        bool success;
+        bytes memory data;
+        for (uint256 i; i < numRegistries; i++) {
+            curveRegistryAddress = curveAddressesProvider.get_address(i);
```

Comment:
Don't think every address from the address provider is necessarily a registry, it's only addresses at positions 0 and 5 from memory

Reply:
Updated to query the 0th and 5th.